server_tgi.py

```diff
@@ -0,0 +1,310 @@
+import argparse
+import json
+import logging
+import time
+import uuid
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import AsyncGenerator, Optional, Tuple
+
+import docker
+import requests
+import uvicorn
+from fastapi import FastAPI, Request
+from fastapi.responses import StreamingResponse
+from huggingface_hub import InferenceClient
+from transformers import AutoTokenizer
+
+from functionary.inference_stream import generate_openai_format_from_stream_async
+from functionary.openai_types import (
+    ChatCompletionChunk,
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseChoice,
+    ChatMessage,
+    Function,
+    StreamChoice,
+    Tool,
+    UsageInfo,
+)
+from functionary.prompt_template import get_prompt_template_from_tokenizer
+from functionary.prompt_template.prompt_utils import enforce_tool_choice
+
+
+def check_health():
+    try:
+        response = requests.get(f"{args.endpoint}/health")
+        if response.status_code == 200:
+            return True
+    except requests.RequestException:
+        pass
+
+    return False
+
+
+def check_hf_model_exists(model_name):
+    url = f"https://huggingface.co/api/models/{model_name}"
+    response = requests.get(url)
+    if response.status_code == 200:
+        return True
+    else:
+        return False
```

Comment:
This would cause the model to not work offline. Also, I find this a bit hacky. Any other way?

Reply:
We may check if the model exists in the container's data directory and load if exists, or load from HF if not?